### PR TITLE
Update Provider.checkFileUrl to allow non-URL file types (like Arweave)

### PR DESCRIPTION
**Pull request by MantisClone**
Required by https://github.com/oceanprotocol/market/pull/1732

Changes proposed in this PR:

- Update `Provider.checkFileUrl` API to take `UrlFile | Arweave` objects instead of a `string` which assumes type url.
- Update Provider test

### CODE REVIEW

**Comment by MantisClone** [comment]
Hello @alexcos20 @kremalicious @mihaisc @bogdanfazakas 
I believe this PR is ready for your review.

CC @AnaLoznianu

File: .github/workflows/ci.yml
Comment (kremalicious) [inline .github/workflows/ci.yml:184]: please don't uncomment this. We know this will fail and it's fine. It's not testing anything

File: test/integration/Provider.test.ts
Comment (kremalicious) [inline test/integration/Provider.test.ts:28]: this should now check for all possible filetypes, not just URLs. So how about an arweave file test here?

File: src/services/Provider.ts
Comment (kremalicious) [inline src/services/Provider.ts:198]: this method name does not make sense anymore if it checks more than URLs. Based on what method does, `getFileInfo()` might be more fitting name?

File: src/services/Provider.ts
Comment (kremalicious) [inline src/services/Provider.ts:199]: why only adding `Arweave`? There are many more types in `@types/Asset` so not clear why only one of those is added.

File: .github/workflows/ci.yml
Comment (MantisClone) [reply .github/workflows/ci.yml]: Done.

File: test/integration/Provider.test.ts
Comment (MantisClone) [reply test/integration/Provider.test.ts:28]: Testing all possible file types is out of scope for this PR, but I can add a test for Arweave.

Comment (MantisClone) [reply test/integration/Provider.test.ts]: Done.

File: src/services/Provider.ts
Comment (MantisClone) [reply src/services/Provider.ts]: Done.

File: src/services/Provider.ts
Comment (MantisClone) [reply src/services/Provider.ts:199]: I only added Arweave in the interest of keeping this PR as small and focused as possible.

**Comment by MantisClone** [comment]
@kremalicious Thank you so much for your review!  I have gone through and tried to address all of your concerns and I believe this PR is ready again for review.  

CC @mihaisc @bogdanfazakas @AnaLoznianu

**Review by kremalicious — APPROVED** [review]
good to go after lint fix, thanks for the patience!

File: test/integration/Provider.test.ts
Comment (MantisClone) [reply test/integration/Provider.test.ts:44]: Done.

**Comment by MantisClone** [comment]
Thanks, @kremalicious for the quick 2nd review!  I am unable to merge this myself because I do not have write access to the repo, so I invite you or any other core team member to merge it on my behalf.
image

File: test/integration/Provider.test.ts
Comment (kremalicious) [reply test/integration/Provider.test.ts:44]: did you push? Don't see a new commit for this or a new test run trigger

Comment (kremalicious) [reply test/integration/Provider.test.ts:44]: bump, as no change was pushed at all so not done

Comment (MantisClone) [reply test/integration/Provider.test.ts]: My bad. It's pushed now.

**Pull request merged**
